### PR TITLE
Sprint 004 B3: Concurrent repairs + cascade-tick span links

### PR DESCRIPTION
**Pull request by riserrad**
## Why this change

> The elegant emergent property — Expert N=1 *is* the single-commitment shape.

(researcher §3, v2 spec)

The mechanic is **N concurrent repair slots**, where N is the per-difficulty WIP cap from `DifficultyTuning.For(difficulty).RepairsPerCycle` — Tutorial 4, Normal 3, Hard 2, Expert 1. Pressing R on the selected subsystem starts a repair if a slot is free; rejects with `All hands busy.` otherwise. Each repair occupies its slot for 1-3 cycles by damage tier and completes back into the existing telemetry surface. Bonded #9 fold-in adds an `ActivityLink` from each `CascadeCheck` Activity back to the `SubsystemTick` that triggered it.

The slot abstraction collapses the v1 dual-limit (per-cycle quota + queue capacity) into a single source of truth — N hands, each occupied for the duration of its repair. Harder difficulty means fewer hands means tighter strategic decisions: at Expert N=1, every R-press is a hard commitment. The slot count tunes both gameplay tempo and commitment density through one knob.

## Impact if not merged

Players stay locked into immediate-only repair, the OTel teaching beat for multi-cycle Activity lifetime + `ActivityLink` (past work + same-cycle causality) doesn't ship, and Sprint 004 close-out blocks. Bonded ship is intentional: splitting B3 from #9 would leave half the link-primitive lesson surface in place without the other half.

## Behavior

- **Press R**: start a repair on the selected subsystem if a slot is free (and that subsystem isn't already under repair). Cycles required = `Math.Clamp((100 - Health) / 33, 1, 3)`, locked at start. Player message: `Started repair on {sub}.`
- **Press R when saturated**: reject with `All hands busy.`; `Telemetry.RepairsDenied` increments with `subsystem.name` tag. The `RepairAction` Activity that `BeginRepair` started is stopped immediately with `status=Error, "rejected"` so it doesn't leak across cycles.
- **Press C**: cancel the oldest in-flight repair on the selected subsystem. Emits a `RepairCancelled` span event with `cancellation.reason="player_cancel"`, sets `status=Error, "cancelled"`, stops the Activity, and increments `Telemetry.RepairsFailed` with `cancellation.reason` for queryable counter dimensionality. Player messages: `Cancelled repair on {sub}.` / `Nothing to cancel on {sub}.`
- Each `StationCycle` while a repair is in flight carries an outgoing `ActivityLink` to that repair's context. Each `CascadeCheck` carries an outgoing link to the source `SubsystemTick` (parent-of would falsify parentage; a link is the right primitive).

## Mechanics

- **`ActiveRepairs`** (`src/SpaceStationMonitor/ActiveRepairs.cs`) holds the slot pool. Sized at construction from `concurrentRepairs`. No `Capacity` accessor (the cap is constructor-fixed; consumers read `AvailableSlots`). `TryStart` rejects on slot exhaustion OR same-subsystem-already-in-flight. `TryCancelOldestOn(target)` removes the oldest matching entry. `InFlightCount`, `AvailableSlots`, `InFlight`, `DecrementAll`, `DrainCompleted` round out the surface.
- **`InFlightRepair`** record: `(Subsystem, Requested, CyclesRemaining, RepairAction)`. Same shape as the v1 entry; renamed to fit the abstraction.
- **`RepairSystem.BeginRepair`** rolls cycles, starts a long-lived `RepairAction` Activity with `subsystem.name`, `repair.requested`, `repair.cycles_required` tags, returns the `InFlightRepair`. **`CompleteRepair`** runs the unchanged `Repair(...)` retry loop, applies `repair.applied`/`repair.healthy` tags, fires the `RepairLeak` event on partial repair, stops the Activity in `finally`. The leak / RetryStorm pathology is preserved verbatim.
- **`GameLoop`** builds the in-flight link list before `StartActivity` and passes via `links:`. Drain step (`DecrementAll` + `DrainCompleted` + `CompleteRepair` + histogram + log) runs immediately after `StartNewCycle`. Cascade loop links each `CascadeCheck` to `tickActivities[cascade.SourceSubsystem]` (the dict is keyed by `sub.Name`, not `actualTarget.Name` — this preserves the `WrongTargetDegradationStrategy` regression test from `a4a4fe8`).
- **`Station`** ctor parameter renamed `repairsPerCycle` → `concurrentRepairs`. `RepairsRemainingThisCycle` and `TryConsumeRepair()` deleted. `ActiveRepairs` exposed as a public property (sized from `concurrentRepairs`).

## HUD (50-col)

```
[1-4] Select [R] Repair [C] Cancel [E] Emerg Pwr
[Q] Quit  Emerg: 3    Smp: idle
Cycle: 42  |  Score: 1240  |  Hands: 2/4
```

- Footer N+3: `Cycle: {n} | Score: {n} | Hands: {n}/{cap}`. `Sel:` cell dropped (selection lives on the subsystem row's `▶` indicator). `Hands` value renders **Yellow at saturation** (`n == cap`) so the player can glance at the cell and know whether pressing R will succeed without parsing the numbers. **In-flight repair state surfaces only via this cell** — per UI lead's deduplication direction at visual conformance review.
- Footer N+2: emergency power counter + sampler badge. The v1 `Repairs: N` cell is gone (its source `RepairsRemainingThisCycle` was deleted under the slot model).
- Hotkey row: adds `[C] Cancel` between `[R] Repair` and `[E] Emerg Pwr`.

## Telemetry

- `station.repairs.denied` description string updated from `"per-cycle quota"` to `"no free repair slot"`. **Counter name and wire format unchanged.**
- `station.repairs.failed` gains a `cancellation.reason="player_cancel"` tag dimension when emitted from the cancellation path. Existing leak-failure emissions are unchanged (no tag).
- All other Activity sites, span tags, link emission, and the bonded #9 cascade-tick link surface are unchanged.

## Tests

- **`ActiveRepairsTests`** (8): capacity + same-subsystem rules on `TryStart`, slot arithmetic invariant, `DecrementAll` + `DrainCompleted` + `TryCancelOldestOn` semantics.
- **`RepairSystemAsyncTests`** (4 + Theory): `BeginRepair` tags + cycle-roll boundaries (1-3 by damage tier), `CompleteRepair` health apply + Activity stop, leak path with `RepairLeak` event.
- **`GameLoopAsyncRepairTests`** (4): completion after rolled cycles, in-flight `ActivityLink` on `StationCycle` (the bonded #9 canary — `InFlightRepair_AddsLinkToStationCycle`), terminal tags + Activity stop, cancel-error path with `RepairCancelled` event + `cancellation.reason` tag.
- **`GameLoopCascadeLinkTests`** (3): cascade ↔ source-tick link under `NoOp`, the divergent path under `WrongTargetDegradationStrategy` (the `a4a4fe8` regression test), and the no-link path when the source tick is sampled out.
- **`ActivityListenerCollection`**: xUnit `[CollectionDefinition(DisableParallelization = true)]` for the listener-bound tests. .NET `ActivitySource` sampling is OR'd across all registered listeners (any-says-yes wins), so a parallel test's `AllData` listener would invalidate another test's `None` decision. Serialization is the only reliable fix without exposing internals.
- **`StationTests`**: the v1 `RepairsRemainingThisCycle` / `TryConsumeRepair` tests are deleted (the abstraction they encoded is gone). New coverage targeting `ActiveRepairs.AvailableSlots` directly. Test count for this class shrinks slightly — that's correct, the new tests are tighter.
- **`RetryStormStrategyTests`**: the quota-exhaustion test rebound to a local `TryConsumeRetry` closure (the `Repair(...)` `tryConsumeQuota` parameter is preserved; the `Station`-sourced quota callback is gone).

## Test plan

- [x] `dotnet build` clean (0 warnings, 0 errors)
- [x] `dotnet test` green (211 / 211)
- [x] Spot-check: `dotnet test --filter "FullyQualifiedName~ActiveRepairs"` green
- [x] Spot-check: `dotnet test --filter "FullyQualifiedName~Station"` green
- [x] Spot-check: `GameLoopAsyncRepairTests.InFlightRepair_AddsLinkToStationCycle` green (bonded #9 canary)
- [ ] Interactive smoke: at Expert (N=1), press R; press R again with one in flight → `All hands busy.` + Hands cell renders Yellow.
- [ ] Interactive smoke: press R, then C → `Cancelled repair on {sub}.`; verify `RepairCancelled` span event in Aspire with `cancellation.reason="player_cancel"`.
- [ ] Interactive smoke: force a cascade (drop a subsystem below 30); verify `CascadeCheck` activity has an `ActivityLink` to the source `SubsystemTick`. Repeat with `BUG_STRATEGY=WrongTargetDegradation` to verify the regression-test path holds at runtime.

## Out of scope (per researcher v2 §7)

- Renaming `DifficultyTuning.RepairsPerCycle` → `DifficultyTuning.ConcurrentRepairs` (touch radius across B1/B2 surfaces too large; follow-up ticket).
- New `in_flight.count` gauge or other new telemetry surface.
- Changes to `RepairSystem.Repair(...)` retry-loop behaviour (the leak bug is intentional and stays).
- Recomputing `cyclesRequired` mid-repair as the subsystem degrades.
- 80-col HUD redesign (locked decision (C); trigger remains Crew + Tasks).
- Cancellation telemetry from non-player paths (event-engine destroy/disable).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by riserrad** [comment]
### Code review

Found 1 issue:

1. `tickActivities` keyed on `actualTarget.Name` but cascade lookup uses `cascade.SourceSubsystem` (= `sub.Name`). Under `WrongTargetDegradationStrategy`, the bug target subsystem's degradation is redirected (`actualTarget != sub`), so nothing is stored under `sub.Name`. When that subsystem goes critical, `TryGetValue(cascade.SourceSubsystem)` finds nothing and the cascade-to-tick link is silently dropped. Fix: `tickActivities[sub.Name] = tickActivity`.

https://github.com/riserrad/OtelPlayground/blob/4f0d177beeb8b61492fa99afd49abbc3ac9a04e9/src/SpaceStationMonitor/GameLoop.cs#L186-L192

`CascadeEngine` sets `SourceSubsystem: sub.Name` (not `actualTarget.Name`):

https://github.com/riserrad/OtelPlayground/blob/4f0d177beeb8b61492fa99afd49abbc3ac9a04e9/src/SpaceStationMonitor/CascadeEngine.cs#L43-L46

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by riserrad** [comment]
### Code review (bonded second-reviewer pass at a4a4fe8)

No issues found. Checked for bugs, CLAUDE.md compliance, and multi-surface correctness.

**Layer 3 — dict-key fix and regression test:**

Fix is surgical: single site at `GameLoop.cs:191`, keying `tickActivities` by `sub.Name` instead of `actualTarget.Name`. No scope creep.

The regression test drives the divergent-name state correctly: `WrongTargetDegradationStrategy("Oxygen", TimeSpan.Zero)` active, Oxygen at health=10 forces a cascade, asserts that the Oxygen-sourced `CascadeCheck` links to the Power-tagged `SubsystemTick`. Under the broken impl, `tickActivities.TryGetValue("Oxygen")` misses entirely → no link → assertion fails. Under the fix, `tickActivities["Oxygen"] = oxygenTick` (tagged `subsystem.name="Power"` due to the redirect) → link binds → assertion passes. The assertion is a structural link-context match — not a default-value tautology.

**Layer 2 — eye-on items:**

All three ActivityListener-dependent test classes carry `[Collection("ActivityListener-bound")]` (`GameLoopCascadeLinkTests`, `GameLoopAsyncRepairTests`, `RepairSystemAsyncTests`). Scope is narrowly correct — other test classes' parallelism unaffected.

`CompleteRepair` has `finally { activity?.Stop(); }` — Activity always stopped on both success and exception paths. `RepairsFailed` increments inside `Repair()` before throw; the catch block's synthetic-failed-result return does not double-bill.

`tryConsumeQuota: null` on async path: quota paid at enqueue via `TryConsumeRepair()`; the async completion correctly passes `null` so retries don't bill again. RetryStorm counter inflation via `Telemetry.RepairsTotal` still fires per retry inside `Repair()`.

**Layer 1 — Activity lifetime:**

`BeginRepair` Activity: started without `using`, stored in `RepairQueueEntry`, stopped via `CompleteRepair`'s `finally` or the cancel path's explicit `Stop()`. No accidental disposal. Tick Activities are `using`-scoped per SubsystemTick iteration (correct — cycle-scoped spans); their `Context` remains readable after disposal when the cascade loop consumes it.

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by riserrad** [comment]
This looks very confusing.

1. Interface is confusing. What does "Q" mean from a player standpoint? 
2. What is the purpose of enqueuing repairs? Think of it from the abstraction point of view. You're a space station mechanical. Why would you enqueue a repair? Does it make sense? The number of repairs you can make in a cycle is more of an abstraction of "how many repairs can a mechanical do at the same time?"
3. It is not clear why a player would be able to queue only a certain number of repairs and why through so many cycles (e.g., 3)
4. If we have the selector aside to the subsystem name, why do we have a display for the selected subsystem besides the "Q"? We should remove that display and keep only the selector indicator above.
5. Should we start thinking about expanding this UI? Let's involve the ui-design-lead.

Overall I really don't like this deliverable. It needs lots of improvements or we should rethink if we really want this. Let's involve the researcher to investigate if this really adds value to the game and learning Observability.

> Note: this branch also has merge conflicts. If we decide to keep pursuing this feature, let's fix it. Otherwise, let's just abandon.

image

**Comment by riserrad** [comment]
Pivoted per your direction. The mechanic is now **N concurrent repair slots** — N = `DifficultyTuning.For(difficulty).RepairsPerCycle` (Tutorial 4, Normal 3, Hard 2, Expert 1). Pressing R on the selected subsystem starts a repair if a slot is free; rejects with `All hands busy.` otherwise. Each repair takes 1-3 cycles based on damage tier. At Expert N=1 every R-press is a hard commitment — that is the framing the v2 spec calls out as the elegant emergent property of the slot abstraction.

Addressing your specific points:

1. **"What does Q mean?"** — gone. The `Q:` cell is removed from the HUD entirely. The footer N+3 row now reads `Cycle: {n} | Score: {n} | Hands: {n}/{cap}`, which is concrete imagery a player parses without help. `Hands: 1/4` reads "I have four hands, one is working." The cell renders Yellow at saturation so the player can glance and know whether pressing R will succeed.
2. **"Why would you enqueue a repair?"** — you don't. There is no queue. The v1 framing dragged enqueue/dequeue/capacity into the player's mental model and into the code. The reshape collapses that into "N hands, each occupied for the duration of its repair." No queue terminology anywhere — the type is `ActiveRepairs`, the entry is `InFlightRepair`, the methods are `TryStart` / `TryCancelOldestOn` / `InFlightCount` / `AvailableSlots`.
3. **"Why a certain number through so many cycles"** — the slot count is the difficulty knob (Tutorial 4 → Expert 1). Cycles required per repair is `Math.Clamp((100 - Health) / 33, 1, 3)` — fast for light damage, slow for heavy damage. Both are gameplay knobs, not abstraction leaks.
4. **"Why Sel: aside the indicator?"** — agreed, the `Sel:` cell is dropped. Selection lives only on the subsystem row's `▶` indicator now.
5. **UI-design-lead engaged.** The HUD redesign spec (`UI Design/2026-04-29-b3-hud-redesign-spec-final.md`) covers the 50-col envelope, all player-facing strings, the Yellow-at-saturation affordance, and the messaging on the new `[C] Cancel` hotkey. 13 ACs landed in this PR.

Bonded #9 cascade-tick `ActivityLink` is preserved verbatim per researcher AC-16 — the OTel link primitive lesson surface is the reason this pair shipped together. The dict-key fix you reviewed at `a4a4fe8` (cascade lookup keyed by source subsystem, not redirect target) is preserved with its regression test.

Branch was force-pushed onto `feature/sprint-004-b3-async-repair-queue` (one push, conflicts and reshape and telemetry adds all included). New head SHA: `7e5c891`. PR title and body rewritten to match.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

**Comment by riserrad** [comment]
### Code review

No issues found. Checked for bugs and CLAUDE.md compliance.

🤖 Generated with [Claude Code](https://claude.ai/code)

**Comment by riserrad** [comment]
### Code review

Found 1 issue:

1. `CancelledRepair_SetsErrorStatus_AndEmitsRepairCancelledEvent` manually re-implements the production cancel Activity sequence (AddEvent → SetStatus → Stop) instead of calling through `HandleCancelRepair`, so the test cannot catch regressions in that code path — e.g., a missing `RepairsFailed.Add` call or reordered operations. (Chester [CLAUDE.md](https://github.com/riserrad/OtelPlayground/blob/20aee565fb1483b5ea7f1f0743932daef6d93adc/.clubhouse/agents/pr-reviewer-chester/CLAUDE.md) says "Shallow tests — tests that verify implementation details rather than behavior")

https://github.com/riserrad/OtelPlayground/blob/20aee565fb1483b5ea7f1f0743932daef6d93adc/tests/SpaceStationMonitor.Tests/GameLoopAsyncRepairTests.cs#L124-L132

---

**`20aee56` patch (GameDisplay.cs, 1+/3−):** clean — removes the redundant `Free: N` cell that duplicated `Hands: N/M` state; column budget comment updated correctly. No new issues.

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by riserrad** [comment]
### Code review (rev2) — c69df19

All rev1 findings addressed. APPROVE.

**Verified at `c69df194c9604f65095198f6078971d31367421a`:**

- **(a) Activity.Current restore** — [RepairSystem.cs:42-50](https://github.com/riserrad/OtelPlayground/blob/c69df194c9604f65095198f6078971d31367421a/src/SpaceStationMonitor/RepairSystem.cs#L40-L52): previousCurrent captured, Current nulled before StartActivity, restored before return. RepairAction is a true root span; no parentage risk.

- **(b) CLAUDE.md hierarchy** — now shows two separate root trees. RepairAction listed as its own root with `↑ each StationCycle while in-flight links to this Activity` annotation. Prose explains cycle→repair causality without parent-of falsification.

- **(c) Cancellation test blocker** — old manual-replay test removed. `HandleCancelRepair_DrivesProductionPath_EmitsExpectedTelemetry` calls `loop.HandleCancelRepair()` directly, wires a MeterListener on `station.repairs.failed`, and asserts both `subsystem.name` and `cancellation.reason="player_cancel"` tags. False coverage gap closed.

- **(d) RepairsFailed exception path** — outer catch in `CompleteRepair` now increments the counter.

- **(e) Outer-catch tag asymmetry** — `failure.layer="outer"` KVP present; inner-catch increments remain untagged. Dedup query convention `RepairsFailed{failure.layer != "outer"}` is coherent.

- **(f) Ctrl+C orphan spans** — `RunAsync` finally block iterates `ActiveRepairs.InFlight`, sets status=Error/"shutdown", stops each Activity, increments `RepairsFailed` with `cancellation.reason="shutdown"`.

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by riserrad** [comment]
Make sure to address the open GitHub Copilot comments above.

File: src/SpaceStationMonitor/GameLoop.cs [hunk 1]
```diff
@@ -120,10 +131,46 @@ _repairSystem.Strategy is SamplingBlindSpotStrategy && isBugActive
...
+                            result.SubsystemName, result.Requested, result.Applied);
+                    }
+                    else
+                    {
+                        _logger.LogInformation("Repair applied to {Name}: {Before:F1}% → {After:F1}%",
+                            result.SubsystemName, result.HealthBefore, result.HealthAfter);
+                    }
+
+                    _display.SetRepairMessage(
+                        $"Repaired {result.SubsystemName}: {result.HealthBefore:F0}% → {result.DisplayedAfter:F0}%");
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/GameLoop.cs]: Stale on this revision. The drain logic referenced was reshaped under the slot-based mechanic; the substantive concern around hard-failure differentiation carries forward and is addressed on the open thread at `CompleteRepair`. Closing here.

File: src/SpaceStationMonitor/GameLoop.cs [hunk 2]
```diff
@@ -300,69 +362,54 @@ private void HandleRepair()
...
+        if (sub.Health >= 100)
+        {
+            _display.SetRepairMessage($"{sub.Name} at full health, no repair needed");
+            _display.Render(_station, _selectedSubsystem, _indicator);
+            return;
+        }
+
+        if (_station.RepairQueue.IsFull)
+        {
+            _display.SetRepairMessage("Repair queue full (4); wait for one to complete");
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/GameLoop.cs]: Stale. `RepairQueue` was removed under the slot-based mechanic; the user message is now `All hands busy.` and reads off `ActiveRepairs.AvailableSlots`. No cap constant to drift on. Closing.

File: src/SpaceStationMonitor/GameLoop.cs [hunk 3]
```diff
@@ -138,6 +185,8 @@ _repairSystem.Strategy is SamplingBlindSpotStrategy && isBugActive
                     tickActivity?.SetTag("degradation.rate",
                         Math.Round(actualTarget.BaseDegradationRate * actualTarget.CascadeMultiplier, 2));
 
+                    tickActivities[actualTarget.Name] = tickActivity;
+
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/GameLoop.cs]: Addressed. `tickActivities` is now keyed on `sub.Name` (the strategy-invariant identity) rather than `actualTarget.Name`. Regression coverage under `WrongTargetDegradationStrategy` exercises the bug-target redirection path. Closing.

File: src/SpaceStationMonitor/GameLoop.cs [hunk 4]
```diff
@@ -305,69 +368,63 @@ private void HandleRepair()
...
-            repairActivity?.SetTag("subsystem.name", result.SubsystemName);
-            repairActivity?.SetTag("repair.requested", result.Requested);
-            repairActivity?.SetTag("repair.applied", result.Applied);
-            repairActivity?.SetTag("repair.healthy", result.IsHealthy);
+            Telemetry.RepairsDenied.Add(1,
+                new KeyValuePair<string, object?>("subsystem.name", sub.Name));
+            _logger.LogInformation("Repair denied on {Name}: all hands busy", sub.Name);
+            _display.SetRepairMessage("All hands busy.");
+            _display.Render(_station, _selectedSubsystem, _indicator);
+            return;
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/GameLoop.cs:412]: PR body has been refreshed; queue framing replaced with the slot-based mechanic. `HandleRepair` rejection (`All hands busy.`) matches the body now. Closing.

File: src/SpaceStationMonitor/GameLoop.cs [hunk 5]
```diff
@@ -305,69 +368,63 @@ private void HandleRepair()
...
-                    result.SubsystemName, result.Requested, result.Applied);
-            }
-            else
-            {
-                _logger.LogInformation("Repair applied to {Name}: {Before:F1}% → {After:F1}%",
-                    result.SubsystemName, result.HealthBefore, result.HealthAfter);
-            }
+            Telemetry.RepairsFailed.Add(1,
+                new KeyValuePair<string, object?>("subsystem.name", sub.Name),
+                new KeyValuePair<string, object?>("cancellation.reason", "player_cancel"));
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/GameLoop.cs:437]: Deferred to a follow-up sprint. AC-18 forbids new metrics in this slice; the `cancellation.reason` tag dimension on `station.repairs.failed` covers operator-query needs by reason. If cardinality concern surfaces in operation, splitting into a distinct counter remains straightforward later. Closing.

File: src/SpaceStationMonitor/GameDisplay.cs [hunk 6]
```diff
@@ -156,28 +156,43 @@ public void Render(Station station, int selectedSubsystem, SampleRegimeIndicator
...
+        var inFlight = station.ActiveRepairs.InFlightCount;
+        var capacity = inFlight + station.ActiveRepairs.AvailableSlots;
+        var handsColor = inFlight >= capacity ? ConsoleColor.Yellow : ConsoleColor.White;
+        WritePaddedSegments(
+            ("  Cycle: ", ConsoleColor.Cyan),
+            ($"{station.CycleCount,-4}", ConsoleColor.White),
+            ("|  Score: ", ConsoleColor.Cyan),
+            ($"{station.Score,-7}", ConsoleColor.White),
+            ("|  Hands: ", ConsoleColor.Cyan),
+            ($"{inFlight}/{capacity}", handsColor));
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/GameDisplay.cs:191]: PR body has been refreshed; the HUD section now describes the `Hands: n/cap` cell, and the footer rendering matches. Closing.

File: src/SpaceStationMonitor/RepairSystem.cs [hunk 7]
```diff
@@ -18,9 +18,64 @@ public RepairSystem(IBugStrategy strategy)
...
+            activity?.AddException(ex);
+            activity?.AddEvent(new ActivityEvent("RepairFailed"));
+            return new RepairResult(
+                SubsystemName: entry.Subsystem.Name,
+                Requested: entry.Requested,
+                Applied: 0,
+                HealthBefore: entry.Subsystem.Health,
+                HealthAfter: entry.Subsystem.Health,
+                DisplayedAfter: entry.Subsystem.Health,
+                IsHealthy: false);
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/RepairSystem.cs:97]: The synthetic `RepairResult(Applied=0, IsHealthy=false)` returned on caught exception is the worst-case leak shape. The drain loop's existing `!IsHealthy` branch logs it as a leak; the trace carries a distinct `RepairFailed` event with `Error` status for diagnostic differentiation in the trace surface. Splitting the metric/log shape between hard-failure and leak is a refinement, not a correctness gap. Deferred. Closing.

File: src/SpaceStationMonitor/RepairSystem.cs [hunk 8]
```diff
@@ -18,9 +18,82 @@ public RepairSystem(IBugStrategy strategy)
...
+            activity?.SetStatus(ActivityStatusCode.Error, ex.Message);
+            activity?.AddException(ex);
+            activity?.AddEvent(new ActivityEvent("RepairFailed"));
+            // Counter rides with the RepairFailed event so unexpected exceptions
+            // (NRE, OOM, lifecycle) don't drop silently. Repair's inner catch already
+            // increments on its own !ShouldRetryAfterFailure / quota-denied paths;
+            // re-throws from there land here too, which is acceptable counter
+            // duplication for the symmetric event-and-counter shape.
+            Telemetry.RepairsFailed.Add(1,
+                new KeyValuePair<string, object?>("subsystem.name", entry.Subsystem.Name));
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/RepairSystem.cs]: This is a deliberate asymmetric-tag pattern. Inner increments in `Repair(...)` are untagged (covering common-bail paths: retries-exhausted and quota-denied). The outer increment in `CompleteRepair` catch tags `failure.layer="outer"` so it covers lifecycle-class exceptions (activity tagging, span events, `Stop`) that the inner cannot see. Dashboards dedupe via `RepairsFailed{failure.layer != "outer"}` for unique-attempt counts; lifecycle visibility is preserved via `RepairsFailed{failure.layer == "outer"}`. The code comment in `CompleteRepair` documents the pattern. Removing the outer increment would erase the lifecycle signal. Closing.

**Comment by riserrad** [comment]
### Code review

No issues found. Checked for bugs and CLAUDE.md compliance.

Verified the 4 diff-verifiable criteria on append commit `4e8ead0`:

1. [`Telemetry.cs`](https://github.com/riserrad/OtelPlayground/blob/4e8ead097627b085294fc56be85be49319d2c357/src/SpaceStationMonitor/Telemetry.cs#L22-L24) — `RepairsDenied` description broadened to cover slot rejection and retry-quota exhaustion; counter name, key, and unit unchanged.

2. [`RepairSystem.cs`](https://github.com/riserrad/OtelPlayground/blob/4e8ead097627b085294fc56be85be49319d2c357/src/SpaceStationMonitor/RepairSystem.cs#L35-L52) — `Activity.Current = previousCurrent` restore is in `finally`; `StartActivity` and all 3 `SetTag` calls are inside `try`.

3. [`GameLoopAsyncRepairTests.cs` cancel-path test](https://github.com/riserrad/OtelPlayground/blob/4e8ead097627b085294fc56be85be49319d2c357/tests/SpaceStationMonitor.Tests/GameLoopAsyncRepairTests.cs#L118-L134) — `MeterListener` callback guards on both `subsystem.name == "Oxygen"` and `cancellation.reason == "player_cancel"` before incrementing.

4. [`GameLoopAsyncRepairTests.cs` no-flight test](https://github.com/riserrad/OtelPlayground/blob/4e8ead097627b085294fc56be85be49319d2c357/tests/SpaceStationMonitor.Tests/GameLoopAsyncRepairTests.cs#L187-L200) — `MeterListener` callback guards on `subsystem.name == "Oxygen"`.

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

File: src/SpaceStationMonitor/ActiveRepairs.cs [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    private readonly List<InFlightRepair> _entries = new();
+
+    public ActiveRepairs(int concurrentRepairs)
+    {
+        _maxConcurrentRepairs = concurrentRepairs;
+    }
+
+    public int InFlightCount => _entries.Count;
+    public int AvailableSlots => _maxConcurrentRepairs - _entries.Count;
+    public IReadOnlyList<InFlightRepair> InFlight => _entries;
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/ActiveRepairs.cs:23]: `InFlight` returns `IReadOnlyList<InFlightRepair>` which is the standard C# convention for exposing an internal collection without ownership transfer. All callers in this codebase consume the property through the interface contract. Cast-back-and-mutate is a hostile-caller concern that does not apply to the internal-only consumption pattern here. Wrapping with `AsReadOnly()` or returning a defensive copy is belt-and-suspenders defense for a non-public API; keeping the convention. Closing.

File: src/SpaceStationMonitor/ActiveRepairs.cs [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    int CyclesRemaining,
+    Activity? RepairAction);
+
+public sealed class ActiveRepairs
+{
+    private readonly int _maxConcurrentRepairs;
+    private readonly List<InFlightRepair> _entries = new();
+
+    public ActiveRepairs(int concurrentRepairs)
+    {
```
[the remark this answers is not in the document]
Comment (riserrad) [reply src/SpaceStationMonitor/ActiveRepairs.cs:17]: `ActiveRepairs` is constructed with `Station(repairsPerCycle)` where `repairsPerCycle` is sourced from `DifficultyTuning.For(difficulty).RepairsPerCycle`. The value range across difficulty tiers is `{1, 2, 3, 4}`. The ctor invariant (`>= 1`) is enforced upstream at the value source rather than at the boundary. A ctor-level `ArgumentOutOfRangeException` guard would be defensive coding for a parameter that cannot reach an out-of-range state through any production callsite. If a callsite is ever added that does not source from `DifficultyTuning`, the guard becomes load-bearing and we add it then. Closing.
